### PR TITLE
[DNM] raftstore: introduce localstate

### PR DESCRIPTION
**Pull request by BusyJay**
This pr merge 4 different meta key into one making meta key management more simple and convenient. And it also introduce `Apply` state, which will be useful when doing async apply.

### CODE REVIEW

File: src/raftstore/store/peer_storage.rs [hunk 1]
```diff
@@ -54,12 +51,8 @@ pub struct PeerStorage {
     engine: Arc<DB>,
 
     pub region: metapb::Region,
```
Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: local_state contains region,  do we need this now?

Comment (BusyJay) [inline src/raftstore/store/peer_storage.rs]: No, local_state only contains raft-related fields.

Comment (siddontang) [inline src/raftstore/store/peer_storage.rs]: oh yes, my fault.

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**